packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts

```diff
@@ -131,11 +131,18 @@ export class ClusterResourceHandler extends ResourceHandler {
     }
 
     if (updates.updateLogging || updates.updateAccess) {
-      const updateResponse = await this.eks.updateClusterConfig({
+      const config: aws.EKS.UpdateClusterConfigRequest = {
         name: this.clusterName,
         logging: this.newProps.logging,
-        resourcesVpcConfig: this.newProps.resourcesVpcConfig,
-      });
+      };
+      if (updates.updateAccess) {
+        config.resourcesVpcConfig = {
```

Comment:
why not `config.resorucesVpcConfig = this.newProps.resourcesVpcConfig`?

Reply:
there was bug in the handler code here because you cannot modify the fields `securityGroupIds` and `subnetIds` (as specified in the warning [here](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/eks/update-cluster-config.html)), and therefore even you pass the value from current state this will fail

Reply:
Ok good to know. Can you add a comment to the code which explains it with the reference?